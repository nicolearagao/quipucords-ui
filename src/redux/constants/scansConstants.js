const ADD_SCAN = 'ADD_SCAN';
const ADD_START_SCAN = 'ADD_START_SCAN';
const UPDATE_SCANS = 'UPDATE_SCANS';
const DELETE_SCAN = 'DELETE_SCAN';

const GET_SCANS = 'GET_SCANS';
const GET_SCAN_CONNECTION_RESULTS = 'GET_SCAN_CONNECTION_RESULTS';
const GET_SCAN_INSPECTION_RESULTS = 'GET_SCAN_INSPECTION_RESULTS';
const GET_SCAN_JOB = 'GET_SCAN_JOB';
const GET_SCAN_JOBS = 'GET_SCAN_JOBS';

const START_SCAN = 'START_SCAN';
const CANCEL_SCAN = 'CANCEL_SCAN';
const PAUSE_SCAN = 'PAUSE_SCAN';
const RESTART_SCAN = 'RESTART_SCAN';

const EDIT_SCAN_HIDE = 'EDIT_SCAN_HIDE';
const EDIT_SCAN_SHOW = 'EDIT_SCAN_SHOW';
const MERGE_SCAN_DIALOG_HIDE = 'MERGE_SCAN_DIALOG_HIDE';
const MERGE_SCAN_DIALOG_SHOW = 'MERGE_SCAN_DIALOG_SHOW';

const SELECT_SCAN = 'SELECT_SCAN';
const DESELECT_SCAN = 'DESELECT_SCAN';
const EXPANDED_SCAN = 'EXPANDED_SCAN';
const NOT_EXPANDED_SCAN = 'NOT_EXPANDED_SCAN';

const RESET_ACTIONS = 'RESET_SCAN_ACTIONS';

const scansTypes = {
  ADD_SCAN,
  ADD_START_SCAN,
  UPDATE_SCANS,
  DELETE_SCAN,
  GET_SCANS,
  GET_SCAN_CONNECTION_RESULTS,
  GET_SCAN_INSPECTION_RESULTS,
  GET_SCAN_JOB,
  GET_SCAN_JOBS,
  START_SCAN,
  CANCEL_SCAN,
  PAUSE_SCAN,
  RESTART_SCAN,
  EDIT_SCAN_HIDE,
  EDIT_SCAN_SHOW,
  MERGE_SCAN_DIALOG_HIDE,
  MERGE_SCAN_DIALOG_SHOW,
  SELECT_SCAN,
  DESELECT_SCAN,
  EXPANDED_SCAN,
  NOT_EXPANDED_SCAN,
  RESET_ACTIONS
};

export {
  scansTypes as default,
  scansTypes,
  ADD_SCAN,
  ADD_START_SCAN,
  UPDATE_SCANS,
  DELETE_SCAN,
  GET_SCANS,
  GET_SCAN_CONNECTION_RESULTS,
  GET_SCAN_INSPECTION_RESULTS,
  GET_SCAN_JOB,
  GET_SCAN_JOBS,
  START_SCAN,
  CANCEL_SCAN,
  PAUSE_SCAN,
  RESTART_SCAN,
  EDIT_SCAN_HIDE,
  EDIT_SCAN_SHOW,
  MERGE_SCAN_DIALOG_HIDE,
  MERGE_SCAN_DIALOG_SHOW,
  SELECT_SCAN,
  DESELECT_SCAN,
  EXPANDED_SCAN,
  NOT_EXPANDED_SCAN,
  RESET_ACTIONS
};
