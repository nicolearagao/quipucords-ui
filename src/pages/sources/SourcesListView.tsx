/**
 * SourcesListView Component
 *
 * This component provides a view for managing sources, including adding, editing, and deleting sources,
 * initiating scans, and displaying source-related information.
 *
 * @module SourcesListView
 */
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import {
  ConditionalTableBody,
  FilterType,
  useTablePropHelpers,
  useTableState
} from '@mturley-latest/react-table-batteries';
import {
  Alert,
  AlertActionCloseButton,
  AlertGroup,
  AlertVariant,
  Button,
  ButtonVariant,
  DropdownItem,
  EmptyState,
  EmptyStateIcon,
  List,
  ListItem,
  Modal,
  ModalVariant,
  PageSection,
  Title,
  ToolbarContent,
  ToolbarItem,
  getUniqueId
} from '@patternfly/react-core';
import { CubesIcon } from '@patternfly/react-icons';
import ActionMenu from 'src/components/ActionMenu';
import { SimpleDropdown } from 'src/components/SimpleDropdown';
import { API_SOURCES_LIST_QUERY } from 'src/constants/apiConstants';
import useSourceApi from 'src/hooks/api/useSourceApi';
import useAlerts from 'src/hooks/useAlerts';
import useQueryClientConfig from 'src/services/queryClientConfig';
import { helpers } from '../../common';
import { ContextIcon, ContextIconVariant } from '../../components/contextIcon/contextIcon';
import { i18nHelpers } from '../../components/i18n/i18nHelpers';
import { RefreshTimeButton } from '../../components/refreshTimeButton/RefreshTimeButton';
import { CredentialType, SourceType } from '../../types';
import AddSourceModal from './components/AddSourceModal';
import { ConnectionsModal } from './components/ConnectionsModal';
import SourcesScanModal from './components/SourcesScanModal';
import { SOURCES_LIST_QUERY, useSourcesQuery } from './useSourcesQuery';

const SourceTypeLabels = {
  acs: 'RHACS',
  ansible: 'Ansible Controller',
  network: 'Network',
  openshift: 'OpenShift',
  satellite: 'Satellite',
  vcenter: 'vCenter Server'
};

const SourcesListView: React.FunctionComponent = () => {
  const { t } = useTranslation();
  const [refreshTime, setRefreshTime] = React.useState<Date | null>();
  const [credentialsSelected, setCredentialsSelected] = React.useState<CredentialType[]>([]);
  const {
    runScan,
    addSource,
    submitEditedSource,
    deleteSource,
    showConnections,
    onEditSource,
    onScanSources,
    scanSelected,
    setScanSelected,
    onDeleteSelectedSources,
    onCloseConnections,
    onScanSource,
    addSourceModal,
    setAddSourceModal,
    setConnectionsData,
    sourceBeingEdited,
    setSourceBeingEdited,
    pendingDeleteSource,
    setPendingDeleteSource,
    connectionsData,
    connectionsSelected,
    setConnectionsSelected
  } = useSourceApi();
  const { queryClient } = useQueryClientConfig();
  const { alerts, addAlert, removeAlert } = useAlerts();
  const { getTimeDisplayHowLongAgo } = helpers;

  /**
   * Invalidates the query cache for the sources list, triggering a refresh.
   */
  const onRefresh = () => {
    queryClient.invalidateQueries({ queryKey: [API_SOURCES_LIST_QUERY] });
  };

  /**
   * Deletes the pending source and handles success, error, and cleanup operations.
   */
  const onDeleteSource = () => {
    deleteSource()
      .then(() => {
        addAlert(
          `Source "${pendingDeleteSource?.name}" deleted successfully`,
          'success',
          getUniqueId()
        );
        onRefresh();
      })
      .catch(err => {
        console.log(err);
        let errorDetail = '';
        if (err?.response?.data) {
          errorDetail += JSON.stringify(err.response.data);
        }
        addAlert(
          `Error removing source ${pendingDeleteSource?.name}. ${errorDetail}`,
          'danger',
          getUniqueId()
        );
      })
      .finally(() => setPendingDeleteSource(undefined));
  };

  /**
   * Submits edited source data, handles success, error, and cleanup operations.
   *
   * @param {SourceType} payload - The payload containing updated source information.
   */
  const onSubmitEditedSource = (payload: SourceType) => {
    submitEditedSource(payload)
      .then(() => {
        addAlert(`${payload.name} updated successfully`, 'success', getUniqueId());
        queryClient.invalidateQueries({ queryKey: [SOURCES_LIST_QUERY] });
        setSourceBeingEdited(undefined);
      })
      .catch(err => {
        console.error({ err });
        addAlert(
          `There was a problem while editing your source. ${JSON.stringify(err?.response?.data)}`,
          'danger',
          getUniqueId()
        );
      });
  };

  /**
   * Initiates a scan for the provided source, handles success, error, and cleanup operations.
   *
   * @param {SourceType} payload - The payload containing source information to start the scan.
   */
  const onRunScan = (payload: SourceType) => {
    runScan(payload)
      .then(() => {
        addAlert(`${payload.name} started to scan`, 'success', getUniqueId());
        queryClient.invalidateQueries({ queryKey: [SOURCES_LIST_QUERY] });
        setScanSelected(undefined);
      })
      .catch(err => {
        console.error({ err });
        addAlert(
          `Error starting scan. ${JSON.stringify(err?.response?.data)}`,
          'danger',
          getUniqueId()
        );
      });
  };

  /**
   * Adds a new source using the provided payload, handles success, error, and cleanup operations.
   *
   * @param {SourceType} payload - The payload containing information for the new source to be added.
   */
  const onAddSource = (payload: SourceType) => {
    addSource(payload)
      .then(() => {
        addAlert(`${payload.name} added successfully`, 'success', getUniqueId());
        queryClient.invalidateQueries({ queryKey: [SOURCES_LIST_QUERY] });
        setAddSourceModal(undefined);
      })
      .catch(err => {
        console.error({ err });
        addAlert(
          `Error adding source. ${JSON.stringify(err?.response?.data)}`,
          'danger',
          getUniqueId()
        );
      });
  };

  /**
   * Retrieves connection information related to a source, updates connection data, and selects the source.
   *
   * @param {SourceType} source - The source for which to retrieve connection information.
   */
  const onShowConnections = (source: SourceType) => {
    showConnections(source)
      .then(res => {
        setConnectionsData({
          successful: res.data.results.filter((c: { status: string }) => c.status === 'success'),
          failure: res.data.results.filter((c: { status: string }) => c.status === 'failure'),
          unreachable: res.data.results.filter(
            (c: { status: string }) => !['success', 'failure'].includes(c.status)
          )
        });
      })
      .catch(err => console.error(err));
    setConnectionsSelected(source);
  };

  const tableState = useTableState({
    persistTo: 'urlParams',
    columnNames: {
      name: 'Name',
      connection: 'Last connected',
      type: 'Type',
      credentials: 'Credentials',
      unreachableSystems: 'Unreachable systems',
      scan: ' ',
      actions: ' '
    },
    filter: {
      isEnabled: true,
      filterCategories: [
        {
          key: 'search_by_name',
          title: 'Name',
          type: FilterType.search,
          placeholderText: 'Filter by name'
        },
        {
          key: 'search_credentials_by_name',
          title: 'Credential name',
          type: FilterType.search,
          placeholderText: 'Filter by credential'
        },
        {
          key: 'source_type',
          title: 'Source type',
          type: FilterType.select,
          placeholderText: 'Filter by source type',
          selectOptions: [
            {
              key: 'ansible',
              label: 'ansible',
              value: 'Ansible'
            },
            {
              key: 'network',
              label: 'network',
              value: 'Network'
            },
            {
              key: 'openshift',
              label: 'openShift',
              value: 'Openshift'
            },
            {
              key: 'rhacs',
              label: 'rhacs',
              value: 'RHACS'
            },
            {
              key: 'satellite',
              label: 'satellite',
              value: 'Satellite'
            },
            {
              key: 'vcenter',
              label: 'vcenter',
              value: 'vCenter'
            }
          ]
        }
      ]
    },
    sort: {
      isEnabled: true,
      sortableColumns: ['name', 'connection', 'type'],
      initialSort: { columnKey: 'name', direction: 'asc' }
    },
    pagination: { isEnabled: true },
    selection: { isEnabled: true }
  });

  const { isLoading, data } = useSourcesQuery({ tableState, setRefreshTime });

  let totalResults = data?.count || 0;
  if (helpers.DEV_MODE) {
    totalResults = helpers.devModeNormalizeCount(totalResults);
  }

  const tableBatteries = useTablePropHelpers({
    ...tableState,
    idProperty: 'id',
    isLoading,
    currentPageItems: data?.results || [],
    totalItemCount: totalResults
  });

  const {
    selection: { selectedItems },
    currentPageItems,
    numRenderedColumns,
    components: {
      Toolbar,
      FilterToolbar,
      PaginationToolbarItem,
      Pagination,
      Table,
      Tbody,
      Td,
      Th,
      Thead,
      Tr
    }
  } = tableBatteries;

  const renderToolbar = () => (
    <Toolbar>
      <ToolbarContent>
        <FilterToolbar id="client-paginated-example-filters" />
        <ToolbarItem>
          <SimpleDropdown
            label="Add Source"
            variant="primary"
            dropdownItems={[
              'Network range',
              'OpenShift',
              'RHACS',
              'Satellite',
              'vCenter server',
              'Ansible controller'
            ].map(type => (
              <DropdownItem key={type} onClick={() => setAddSourceModal(type)}>
                {type}
              </DropdownItem>
            ))}
          />
        </ToolbarItem>
        <ToolbarItem>
          <Button
            variant={ButtonVariant.secondary}
            isDisabled={Object.values(selectedItems).filter(val => val !== null).length <= 1}
            onClick={onScanSources}
          >
            {t('table.label', { context: 'scan' })}
          </Button>
        </ToolbarItem>
        <ToolbarItem>
          <Button
            variant={ButtonVariant.secondary}
            isDisabled={!selectedItems?.length}
            onClick={onDeleteSelectedSources}
          >
            {t('table.label', { context: 'delete' })}
          </Button>
        </ToolbarItem>
        <ToolbarItem>
          <RefreshTimeButton lastRefresh={refreshTime?.getTime() ?? 0} onRefresh={onRefresh} />
        </ToolbarItem>
        <PaginationToolbarItem>
          <Pagination variant="top" isCompact widgetId="client-paginated-example-pagination" />
        </PaginationToolbarItem>
      </ToolbarContent>
    </Toolbar>
  );

  const renderConnection = (source: SourceType): React.ReactNode => {
    if (!source?.connection) {
      return null;
    }
    const isPending =
      source.connection.status === 'created' ||
      source.connection.status === 'pending' ||
      source.connection.status === 'running';
    const scanTime = (isPending && source.connection.start_time) || source.connection.end_time;

    const statusString = i18nHelpers.translate(t, 'table.label', {
      context: ['status', source.connection.status, 'sources']
    });
    return (
      <Button
        variant={ButtonVariant.link}
        onClick={() => {
          onShowConnections(source);
        }}
      >
        <ContextIcon symbol={ContextIconVariant[source.connection.status]} /> {statusString}{' '}
        {getTimeDisplayHowLongAgo(scanTime)}
      </Button>
    );
  };

  return (
    <PageSection variant="light">
      {renderToolbar()}
      <Table aria-label="Example things table" variant="compact">
        <Thead>
          <Tr isHeaderRow>
            <Th columnKey="name" />
            <Th columnKey="connection" />
            <Th columnKey="type" />
            <Th columnKey="credentials" />
            <Th columnKey="scan" />
            <Th columnKey="actions" />
          </Tr>
        </Thead>
        <ConditionalTableBody
          isLoading={isLoading}
          isNoData={currentPageItems.length === 0}
          noDataEmptyState={
            <EmptyState variant="sm">
              <EmptyStateIcon icon={CubesIcon} />
              <Title headingLevel="h2" size="lg">
                No sources available
              </Title>
            </EmptyState>
          }
          numRenderedColumns={numRenderedColumns}
        >
          <Tbody>
            {currentPageItems?.map((source: SourceType, rowIndex) => (
              <Tr key={source.id} item={source} rowIndex={rowIndex}>
                <Td columnKey="name">{source.name}</Td>
                <Td columnKey="connection">{renderConnection(source)}</Td>
                <Td columnKey="type">{SourceTypeLabels[source.source_type]}</Td>
                <Td columnKey="credentials">
                  <Button
                    variant={ButtonVariant.link}
                    onClick={() => {
                      setCredentialsSelected(source.credentials);
                    }}
                  >
                    {source.credentials.length}
                  </Button>
                </Td>
                <Td isActionCell columnKey="scan">
                  <Button
                    isDisabled={source.connection?.status === 'pending'}
                    variant={ButtonVariant.link}
                    onClick={() => onScanSource(source)}
                  >
                    Scan
                  </Button>
                </Td>
                <Td isActionCell columnKey="actions">
                  <ActionMenu<SourceType>
                    item={source}
                    actions={[
                      { label: 'Edit', onClick: onEditSource },
                      { label: 'Delete', onClick: setPendingDeleteSource }
                    ]}
                  />
                </Td>
              </Tr>
            ))}
          </Tbody>
        </ConditionalTableBody>
      </Table>
      <Pagination variant="bottom" widgetId="server-paginated-example-pagination" />
      {!!credentialsSelected.length && (
        <Modal
          variant={ModalVariant.small}
          title="Credentials"
          isOpen={!!credentialsSelected}
          onClose={() => setCredentialsSelected([])}
          actions={[
            <Button key="cancel" variant="secondary" onClick={() => setCredentialsSelected([])}>
              Close
            </Button>
          ]}
        >
          <List isPlain isBordered>
            {credentialsSelected.map(c => (
              <ListItem key={c.name}>{c.name}</ListItem>
            ))}
          </List>
        </Modal>
      )}
      {connectionsSelected && (
        <ConnectionsModal
          source={connectionsSelected}
          connections={connectionsData}
          onClose={onCloseConnections}
        />
      )}

      {!!pendingDeleteSource && (
        <Modal
          variant={ModalVariant.small}
          title="Permanently delete source"
          isOpen={!!pendingDeleteSource}
          onClose={() => setPendingDeleteSource(undefined)}
          actions={[
            <Button key="confirm" variant="danger" onClick={() => onDeleteSource()}>
              Delete
            </Button>,
            <Button key="cancel" variant="link" onClick={() => setPendingDeleteSource(undefined)}>
              Cancel
            </Button>
          ]}
        >
          Are you sure you want to delete the source &quot;
          {pendingDeleteSource.name}&quot;
        </Modal>
      )}

      {sourceBeingEdited && (
        <AddSourceModal
          source={sourceBeingEdited}
          type={sourceBeingEdited.source_type}
          onClose={() => setSourceBeingEdited(undefined)}
          onSubmit={onSubmitEditedSource}
        />
      )}
      {addSourceModal && (
        <AddSourceModal
          type={addSourceModal}
          onClose={() => setAddSourceModal(undefined)}
          onSubmit={onAddSource}
        />
      )}

      {scanSelected && (
        <SourcesScanModal
          onClose={() => setScanSelected(undefined)}
          onSubmit={onRunScan}
          sources={scanSelected}
        />
      )}
      <AlertGroup isToast isLiveRegion>
        {alerts.map(({ key, variant, title }) => (
          <Alert
            timeout={8000}
            onTimeout={() => key && removeAlert(key)}
            variant={AlertVariant[variant || 'info']}
            title={title}
            actionClose={
              <AlertActionCloseButton
                title={title as string}
                variantLabel={`${variant} alert`}
                onClose={() => key && removeAlert(key)}
              />
            }
            key={key}
          />
        ))}
      </AlertGroup>
    </PageSection>
  );
};

export default SourcesListView;
