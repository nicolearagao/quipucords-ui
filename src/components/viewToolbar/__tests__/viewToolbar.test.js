import React from 'react';
import { ViewToolbar } from '../viewToolbar';
import { API_QUERY_TYPES } from '../../../constants/apiConstants';
import { CONFIG as sourcesConfig } from '../../sources/sources';

describe('ViewToolbar Component', () => {
  it('should render a basic component', async () => {
    const props = {
      useView: () => ({ viewId: sourcesConfig.viewId, config: { toolbar: sourcesConfig.toolbar } })
    };
    const component = await shallowComponent(<ViewToolbar {...props} />);

    expect(component).toMatchSnapshot('basic');
  });

  it('should hide categories when a single filter is available', () => {
    const props = {
      useView: () => ({
        viewId: sourcesConfig.viewId,
        config: {
          toolbar: { filterFields: [sourcesConfig.toolbar.filterFields[0]] }
        }
      })
    };
    const component = renderComponent(<ViewToolbar {...props} />);

    expect(component.querySelectorAll('input[type="text"]')).toMatchSnapshot('single filter');
  });

  it('should handle updating toolbar chips', () => {
    const props = {
      useView: () => ({
        viewId: sourcesConfig.viewId,
        query: { [API_QUERY_TYPES.SEARCH_NAME]: 'lorem ipsum' },
        config: {
          toolbar: { filterFields: [sourcesConfig.toolbar.filterFields[0]] }
        }
      })
    };
    const component = renderComponent(<ViewToolbar {...props} />);

    expect(component.querySelectorAll('.pf-c-chip')).toMatchSnapshot('chips');
  });

  it('should handle displaying secondary components, fields', async () => {
    const props = {
      useView: () => ({
        viewId: sourcesConfig.viewId,
        config: {
          toolbar: { filterFields: [sourcesConfig.toolbar.filterFields[0]] }
        }
      }),
      secondaryFields: <React.Fragment>dolor sit</React.Fragment>
    };
    const component = await shallowComponent(<ViewToolbar {...props} />);

    expect(component).toMatchSnapshot('secondary');
  });
});
