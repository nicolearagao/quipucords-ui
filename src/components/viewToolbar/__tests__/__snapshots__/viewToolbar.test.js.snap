// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewToolbar Component should handle displaying secondary components, fields: secondary 1`] = `
<React.Fragment>
  <Toolbar
    className="quipucords-toolbar"
    clearAllFilters={[Function]}
    clearFiltersButtonText="t(toolbar.label, {"context":"clear-filters"})"
    collapseListedFiltersBreakpoint="lg"
  >
    <ToolbarContent
      isExpanded={false}
      showClearFiltersButton={false}
    >
      <ToolbarToggleGroup
        breakpoint="lg"
        toggleIcon={
          <FilterIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        }
      >
        <ForwardRef
          variant="filter-group"
        >
          <ToolbarFilter
            categoryName="t(toolbar.label_option, {"context":"search_by_name"})"
            chips={[]}
            deleteChip={[Function]}
            showToolbarItem={true}
          >
            <SearchName />
          </ToolbarFilter>
        </ForwardRef>
      </ToolbarToggleGroup>
      <ToolbarItem
        variant="separator"
      />
      <ToolbarItem
        spacer={
          {
            "default": "spacerSm",
          }
        }
      >
        <ViewToolbarFieldSort
          t={[Function]}
          useOnSelect={[Function]}
          useView={[Function]}
        />
      </ToolbarItem>
      <ToolbarItem
        variant="separator"
      />
      <ToolbarItem>
        <RefreshTimeButton
          lastRefresh={0}
          onRefresh={[Function]}
          t={[Function]}
        />
      </ToolbarItem>
      <ToolbarItem
        alignment={
          {
            "lg": "alignRight",
            "md": "alignLeft",
          }
        }
      >
        <React.Fragment>
          dolor sit
        </React.Fragment>
      </ToolbarItem>
    </ToolbarContent>
  </Toolbar>
  <Divider />
</React.Fragment>
`;

exports[`ViewToolbar Component should handle updating toolbar chips: chips 1`] = `
NodeList [
  <div
    class="pf-c-chip"
    data-ouia-component-id="OUIA-Generated-Chip-1"
    data-ouia-component-type="PF4/Chip"
    data-ouia-safe="true"
  >
    <span
      class="pf-c-chip__text"
      id="pf-random-id-5"
    >
      t(toolbar.label_chip, {"context":"lorem ipsum"})
    </span>
    <button
      aria-disabled="false"
      aria-label="close"
      aria-labelledby="remove_pf-random-id-5 pf-random-id-5"
      class="pf-c-button pf-m-plain"
      data-ouia-component-id="close"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      id="remove_pf-random-id-5"
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 352 512"
        width="1em"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        />
      </svg>
    </button>
  </div>,
]
`;

exports[`ViewToolbar Component should hide categories when a single filter is available: single filter 1`] = `
NodeList [
  <input
    aria-invalid="false"
    aria-label="t(toolbar.label_placeholder_filter, {"context":"search_by_name"})"
    class="pf-c-form-control quipucords-form__text-input "
    data-ouia-component-id="OUIA-Generated-TextInputBase-2"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    data-test="toolbarSearchName"
    id="generatedid-"
    maxlength="255"
    name="generatedid-"
    placeholder="t(toolbar.label_placeholder_filter, {"context":"search_by_name"})"
    type="text"
    value=""
  />,
]
`;

exports[`ViewToolbar Component should render a basic component: basic 1`] = `
<React.Fragment>
  <Toolbar
    className="quipucords-toolbar"
    clearAllFilters={[Function]}
    clearFiltersButtonText="t(toolbar.label, {"context":"clear-filters"})"
    collapseListedFiltersBreakpoint="lg"
  >
    <ToolbarContent
      isExpanded={false}
      showClearFiltersButton={false}
    >
      <ToolbarToggleGroup
        breakpoint="lg"
        toggleIcon={
          <FilterIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        }
      >
        <ForwardRef
          variant="filter-group"
        >
          <ToolbarItem>
            <ViewToolbarSelectCategory
              t={[Function]}
              useOnSelect={[Function]}
              useSelector={[Function]}
              useView={[Function]}
            />
          </ToolbarItem>
          <ToolbarFilter
            categoryName="t(toolbar.label_option, {"context":"search_by_name"})"
            chips={[]}
            deleteChip={[Function]}
            showToolbarItem={false}
          >
            <SearchName />
          </ToolbarFilter>
          <ToolbarFilter
            categoryName="t(toolbar.label_option, {"context":"search_credentials_by_name"})"
            chips={[]}
            deleteChip={[Function]}
            showToolbarItem={false}
          >
            <SearchCredentialsName />
          </ToolbarFilter>
          <ToolbarFilter
            categoryName="t(toolbar.label_option, {"context":"source_type"})"
            chips={[]}
            deleteChip={[Function]}
            showToolbarItem={false}
          >
            <SourceType />
          </ToolbarFilter>
        </ForwardRef>
      </ToolbarToggleGroup>
      <ToolbarItem
        variant="separator"
      />
      <ToolbarItem
        spacer={
          {
            "default": "spacerSm",
          }
        }
      >
        <ViewToolbarFieldSort
          t={[Function]}
          useOnSelect={[Function]}
          useView={[Function]}
        />
      </ToolbarItem>
      <ToolbarItem
        variant="separator"
      />
      <ToolbarItem>
        <RefreshTimeButton
          lastRefresh={0}
          onRefresh={[Function]}
          t={[Function]}
        />
      </ToolbarItem>
      <ToolbarItem
        alignment={
          {
            "lg": "alignRight",
            "md": "alignLeft",
          }
        }
      />
    </ToolbarContent>
  </Toolbar>
  <Divider />
</React.Fragment>
`;
